src/simulation/mujocoDynamics/MJSystemMassMatrix/_UnitTest/test_MJSystemMassMatrix.py

```diff
@@ -0,0 +1,324 @@
+#
+#  ISC License
+#
+#  Copyright (c) 2025, Autonomous Vehicle Systems Lab, University of Colorado Boulder
+#
+#  Permission to use, copy, modify, and/or distribute this software for any
+#  purpose with or without fee is hereby granted, provided that the above
+#  copyright notice and this permission notice appear in all copies.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+#  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+#  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+#  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+#  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+#  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+#  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+#
+
+import os
+import numpy as np
+from numpy.testing import assert_allclose, assert_equal, assert_array_less
+import pytest
+import tempfile
+
+from Basilisk.utilities import SimulationBaseClass, macros
+try:
+    from Basilisk.simulation import MJSystemMassMatrix, mujoco
+
+    couldImportMujoco = True
+except:
+    couldImportMujoco = False
+
+# Constants used throughout tests
+HUB_MASS  = 50.0
+HUB_I_COM = np.diag([10.0, 11.0, 12.0])   # hub inertia @ COM
+ARM_MASS  = 10.0
+ROD_I_COM = np.diag([ 5.0,  6.0,  7.0])   # child inertia @ COM
+SX, SY, SZ = 0.5, 0.3, 0.2
+
+def _write_tmp(xml_text: str, base: str) -> str:
+    td = tempfile.mkdtemp(prefix="mjmass_")
+    p = os.path.join(td, f"{base}.xml")
+    with open(p, "w") as f:
+        f.write(xml_text)
+    return p
+
+def xml_single():
+    return f"""<mujoco>
+  <option gravity="0 0 0"/>
+  <worldbody>
+    <body name="hub">
+      <freejoint/>
+      <inertial pos="0 0 0" quat="1 0 0 0" mass="{HUB_MASS}" diaginertia="{HUB_I_COM[0,0]} {HUB_I_COM[1,1]} {HUB_I_COM[2,2]}"/>
+      <geom type="box" size="{SX} {SY} {SZ}"/>
+    </body>
+  </worldbody>
+</mujoco>"""
```

Comment:
Why functions instead of constant values:
```suggestion
XML_SINGLE = f"""<mujoco>
  <option gravity="0 0 0"/>
  <worldbody>
    <body name="hub">
      <freejoint/>
      <inertial pos="0 0 0" quat="1 0 0 0" mass="{HUB_MASS}" diaginertia="{HUB_I_COM[0,0]} {HUB_I_COM[1,1]} {HUB_I_COM[2,2]}"/>
      <geom type="box" size="{SX} {SY} {SZ}"/>
    </body>
  </worldbody>
</mujoco>"""
```